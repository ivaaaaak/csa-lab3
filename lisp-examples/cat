(set char (read_char))
(while (!= char 0)
	(print_char char)
	(set char (read_char)))