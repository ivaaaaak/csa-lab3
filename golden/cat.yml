source: |-
  (set char (read_char))
  (while (!= char 0)
    (print_char char)
    (set char (read_char)))

input: |-
  foo

code: |-
  0 - D0000002 - jmp 2

  DATA MEMORY
  1 - 00000000 - 0
  
  CODE MEMORY
  2 - 60000000 - input
  3 - 50000001 - save 1
  4 - 40000001 - load 1
  5 - C2000000 - compare #0
  6 - E0000009 - jz 9
  7 - 42000001 - load #1
  8 - D000000A - jmp 10
  9 - 42000000 - load #0
  10 - E0000010 - jz 16
  11 - 40000001 - load 1
  12 - 70000000 - print
  13 - 60000000 - input
  14 - 50000001 - save 1
  15 - D0000004 - jmp 4
  16 - F0000000 - halt

output: |
  source LoC: 4 machine code instr: 17
  ============================================================
  foo
  instr_counter: 41, ticks: 52

log: |
  DEBUG   machine:simulation    TICK:    0, IP:    0, AR:    0, SP: 2048, ALU:    0, ACC:    0 	jmp 2
  DEBUG   machine:simulation    TICK:    1, IP:    2, AR:    0, SP: 2048, ALU:    0, ACC:    0 	input
  DEBUG   machine:simulation    TICK:    2, IP:    3, AR:    0, SP: 2048, ALU:    0, ACC:  102 	save 1
  DEBUG   machine:simulation    TICK:    4, IP:    4, AR:    1, SP: 2048, ALU:    0, ACC:  102 	load 1
  DEBUG   machine:simulation    TICK:    6, IP:    5, AR:    1, SP: 2048, ALU:  102, ACC:  102 	compare #0
  DEBUG   machine:simulation    TICK:    7, IP:    6, AR:    1, SP: 2048, ALU:  102, ACC:  102 	jz 9
  DEBUG   machine:simulation    TICK:    8, IP:    7, AR:    1, SP: 2048, ALU:  102, ACC:  102 	load #1
  DEBUG   machine:simulation    TICK:    9, IP:    8, AR:    1, SP: 2048, ALU:    1, ACC:    1 	jmp 10
  DEBUG   machine:simulation    TICK:   10, IP:   10, AR:    1, SP: 2048, ALU:    1, ACC:    1 	jz 16
  DEBUG   machine:simulation    TICK:   11, IP:   11, AR:    1, SP: 2048, ALU:    1, ACC:    1 	load 1
  DEBUG   machine:simulation    TICK:   13, IP:   12, AR:    1, SP: 2048, ALU:  102, ACC:  102 	print
  DEBUG   data_path:signal_output output: '' << 'f'
  DEBUG   machine:simulation    TICK:   14, IP:   13, AR:    1, SP: 2048, ALU:  102, ACC:  102 	input
  DEBUG   machine:simulation    TICK:   15, IP:   14, AR:    1, SP: 2048, ALU:  102, ACC:  111 	save 1
  DEBUG   machine:simulation    TICK:   17, IP:   15, AR:    1, SP: 2048, ALU:  102, ACC:  111 	jmp 4
  DEBUG   machine:simulation    TICK:   18, IP:    4, AR:    1, SP: 2048, ALU:  102, ACC:  111 	load 1
  DEBUG   machine:simulation    TICK:   20, IP:    5, AR:    1, SP: 2048, ALU:  111, ACC:  111 	compare #0
  DEBUG   machine:simulation    TICK:   21, IP:    6, AR:    1, SP: 2048, ALU:  111, ACC:  111 	jz 9
  DEBUG   machine:simulation    TICK:   22, IP:    7, AR:    1, SP: 2048, ALU:  111, ACC:  111 	load #1
  DEBUG   machine:simulation    TICK:   23, IP:    8, AR:    1, SP: 2048, ALU:    1, ACC:    1 	jmp 10
  DEBUG   machine:simulation    TICK:   24, IP:   10, AR:    1, SP: 2048, ALU:    1, ACC:    1 	jz 16
  DEBUG   machine:simulation    TICK:   25, IP:   11, AR:    1, SP: 2048, ALU:    1, ACC:    1 	load 1
  DEBUG   machine:simulation    TICK:   27, IP:   12, AR:    1, SP: 2048, ALU:  111, ACC:  111 	print
  DEBUG   data_path:signal_output output: 'f' << 'o'
  DEBUG   machine:simulation    TICK:   28, IP:   13, AR:    1, SP: 2048, ALU:  111, ACC:  111 	input
  DEBUG   machine:simulation    TICK:   29, IP:   14, AR:    1, SP: 2048, ALU:  111, ACC:  111 	save 1
  DEBUG   machine:simulation    TICK:   31, IP:   15, AR:    1, SP: 2048, ALU:  111, ACC:  111 	jmp 4
  DEBUG   machine:simulation    TICK:   32, IP:    4, AR:    1, SP: 2048, ALU:  111, ACC:  111 	load 1
  DEBUG   machine:simulation    TICK:   34, IP:    5, AR:    1, SP: 2048, ALU:  111, ACC:  111 	compare #0
  DEBUG   machine:simulation    TICK:   35, IP:    6, AR:    1, SP: 2048, ALU:  111, ACC:  111 	jz 9
  DEBUG   machine:simulation    TICK:   36, IP:    7, AR:    1, SP: 2048, ALU:  111, ACC:  111 	load #1
  DEBUG   machine:simulation    TICK:   37, IP:    8, AR:    1, SP: 2048, ALU:    1, ACC:    1 	jmp 10
  DEBUG   machine:simulation    TICK:   38, IP:   10, AR:    1, SP: 2048, ALU:    1, ACC:    1 	jz 16
  DEBUG   machine:simulation    TICK:   39, IP:   11, AR:    1, SP: 2048, ALU:    1, ACC:    1 	load 1
  DEBUG   machine:simulation    TICK:   41, IP:   12, AR:    1, SP: 2048, ALU:  111, ACC:  111 	print
  DEBUG   data_path:signal_output output: 'fo' << 'o'
  DEBUG   machine:simulation    TICK:   42, IP:   13, AR:    1, SP: 2048, ALU:  111, ACC:  111 	input
  DEBUG   machine:simulation    TICK:   43, IP:   14, AR:    1, SP: 2048, ALU:  111, ACC:    0 	save 1
  DEBUG   machine:simulation    TICK:   45, IP:   15, AR:    1, SP: 2048, ALU:  111, ACC:    0 	jmp 4
  DEBUG   machine:simulation    TICK:   46, IP:    4, AR:    1, SP: 2048, ALU:  111, ACC:    0 	load 1
  DEBUG   machine:simulation    TICK:   48, IP:    5, AR:    1, SP: 2048, ALU:    0, ACC:    0 	compare #0
  DEBUG   machine:simulation    TICK:   49, IP:    6, AR:    1, SP: 2048, ALU:    0, ACC:    0 	jz 9
  DEBUG   machine:simulation    TICK:   50, IP:    9, AR:    1, SP: 2048, ALU:    0, ACC:    0 	load #0
  DEBUG   machine:simulation    TICK:   51, IP:   10, AR:    1, SP: 2048, ALU:    0, ACC:    0 	jz 16
  DEBUG   machine:simulation    TICK:   52, IP:   16, AR:    1, SP: 2048, ALU:    0, ACC:    0 	halt
